File: app/src/test/java/com/iyoa/cleanaddis/unitTesting/viewModel/PostViewModelTest.java
package com.iyoa.cleanaddis.unitTesting.viewModel;

import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
import androidx.room.Room;
import androidx.test.core.app.ApplicationProvider;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.MediumTest;
import kotlinx.coroutines.ExperimentalCoroutinesApi;
import kotlinx.coroutines.runBlocking;
import org.hamcrest.CoreMatchers;
import org.junit.*;
import org.junit.runner.RunWith;

@ExperimentalCoroutinesApi
@RunWith(AndroidJUnit4::class)
@MediumTest
public class PostViewModelTest {
    private lateinit var repo: CommentRepository
    private lateinit var database: PostDatabase

    @get:Rule
    var instantExecutorRule = InstantTaskExecutorRule()

    @Before
    fun setup() {
        database = Room.databaseBuilder(
                ApplicationProvider.getApplicationContext(),
                PostDatabase::class.java,
                "comment").allowMainThreadQueries().build()

        repo = CommentRepository(database.commentDao())
    }
    @After
    fun cleanUp() {
        database.close()
    }
    @Test
    fun addAndRetrieveUsingRepo()= runBlocking{
        val createdPost = Post("h3j23hpost", "edenAb", 7,7,1,1,"co3mm3n4ter","co3media34ter","ALL","POSTED",Date(2019))
        repo.insertPost(createdPost)

        val result  = repo.getPostByUUID("h3j23hpost")

        Assert.assertThat(result, CoreMatchers.notNullValue())
    }

}

File: app/src/test/java/com/iyoa/cleanaddis/unitTesting/viewModel/PostViewModelTest.kt
package com.iyoa.cleanaddis.unitTesting.viewModel


import androidx.arch.core.executor.testing.InstantTaskExecutorRule
import com.google.android.gms.tasks.Task
import com.iyoa.cleanaddis.viewModels.posting.PostViewModel;
import com.google.common.truth.Truth.assertThat
import com.iyoa.cleanaddis.data.posting.PostUUID
import com.iyoa.cleanaddis.entity.posting.Post
import io.reactivex.internal.util.NotificationLite.getValue
import kotlinx.coroutines.ExperimentalCoroutinesApi
import org.junit.Before
import org.junit.Rule
import org.junit.Test
@ExperimentalCoroutinesApi
class PostViewModelTest {
    // Subject under test
    private lateinit var postViewModel : PostViewModel
    private lateinit var post : PostUUID

    // Set the main coroutines dispatcher for unit testing.
    @ExperimentalCoroutinesApi
    @get:Rule
    var mainCoroutineRule = MainCoroutineRule()

    // Executes each task synchronously using Architecture Components.
    @get:Rule var instantExecutorRule = InstantTaskExecutorRule()

    //val post = Post("Title1", "Description1")

    @Before
    fun setupViewModel() {

       // postViewModel = PostViewModel()
    }

    @Test
    fun getActiveTaskFromRepositoryAndLoadIntoView() {
        postViewModel.getResponses

        // Then verify that the view was notified
        assertThat(postViewModel.post.get()!!.uuid).isEqualTo(this.post.uuid)

    }

}
